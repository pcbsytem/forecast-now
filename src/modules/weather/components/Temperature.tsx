import { Col, FlexboxGrid } from "rsuite";
import {
  TemperatureContainer,
  TemperatureListContainer,
  TemperatureText,
} from "../Weather.style";

const Temperature = (temperatureList) => {
  const list = temperatureList
    ? Object.keys(temperatureList).map((key) => temperatureList[key])
    : [];

  return (
    <TemperatureListContainer
      as={Col}
      colspan={24}
      md={2}
      screenWidth={window.screen.availWidth}
    >
      {list.map((listItem, index) => (
        <TemperatureContainer key={index}>
          <TemperatureText>{listItem.temperature}</TemperatureText>
          <b>Â°F</b>
        </TemperatureContainer>
      ))}
    </TemperatureListContainer>
  );
};

export default Temperature;
