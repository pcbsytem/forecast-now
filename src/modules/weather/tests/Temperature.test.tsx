import React from "react";
import { render } from "@testing-library/react";
import Temperature from "../components/Temperature";

const mockWeatherPeriod = {
  "0": {
    number: 14,
    name: "Tuesday Night",
    startTime: "2022-07-05T18:00:00-04:00",
    endTime: "2022-07-06T06:00:00-04:00",
    isDaytime: false,
    temperature: 72,
    temperatureUnit: "F",
    temperatureTrend: null,
    windSpeed: "6 mph",
    windDirection: "W",
    icon: "https://api.weather.gov/icons/land/night/tsra_sct,40/tsra_sct,30?size=medium",
    shortForecast: "Chance Showers And Thunderstorms",
    detailedForecast:
      "A chance of showers and thunderstorms. Mostly cloudy, with a low around 72. Chance of precipitation is 40%.",
  },
  "1": {
    number: 13,
    name: "Tuesday",
    startTime: "2022-07-05T06:00:00-04:00",
    endTime: "2022-07-05T18:00:00-04:00",
    isDaytime: true,
    temperature: 91,
    temperatureUnit: "F",
    temperatureTrend: null,
    windSpeed: "2 to 7 mph",
    windDirection: "S",
    icon: "https://api.weather.gov/icons/land/day/rain_showers,40/tsra_hi,40?size=medium",
    shortForecast: "Chance Rain Showers then Chance Showers And Thunderstorms",
    detailedForecast:
      "A chance of rain showers before 8am, then a chance of showers and thunderstorms. Partly sunny, with a high near 91. Chance of precipitation is 40%.",
  },
};

describe("Temperature component", () => {
  it("Should render correctly", () => {
    expect(render(<Temperature {...mockWeatherPeriod} />)).toMatchSnapshot();
  });

  it("Should render Temperature with empty list", () => {
    expect(render(<Temperature {...mockWeatherPeriod} />)).toMatchSnapshot();
  });
});
