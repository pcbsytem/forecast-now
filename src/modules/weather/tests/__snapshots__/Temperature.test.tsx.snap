// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Temperature component Should render Temperature with empty list 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rs-flex-box-grid-item rs-flex-box-grid-item-24 rs-col rs-col-md-2"
        role="gridcell"
      >
        <div
          class="sc-bczRLJ hnZsda rs-flex-box-grid rs-flex-box-grid-top rs-flex-box-grid-start"
        >
          <div
            class="sc-gsnTZi iiSpRq rs-flex-box-grid-item rs-flex-box-grid-item-0"
          >
            <b
              class="sc-dkzDqf cybjuV"
            >
              72
            </b>
            <b>
              °F
            </b>
          </div>
          <div
            class="sc-gsnTZi iiSpRq rs-flex-box-grid-item rs-flex-box-grid-item-0"
          >
            <b
              class="sc-dkzDqf cybjuV"
            >
              91
            </b>
            <b>
              °F
            </b>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rs-flex-box-grid-item rs-flex-box-grid-item-24 rs-col rs-col-md-2"
      role="gridcell"
    >
      <div
        class="sc-bczRLJ hnZsda rs-flex-box-grid rs-flex-box-grid-top rs-flex-box-grid-start"
      >
        <div
          class="sc-gsnTZi iiSpRq rs-flex-box-grid-item rs-flex-box-grid-item-0"
        >
          <b
            class="sc-dkzDqf cybjuV"
          >
            72
          </b>
          <b>
            °F
          </b>
        </div>
        <div
          class="sc-gsnTZi iiSpRq rs-flex-box-grid-item rs-flex-box-grid-item-0"
        >
          <b
            class="sc-dkzDqf cybjuV"
          >
            91
          </b>
          <b>
            °F
          </b>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Temperature component Should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rs-flex-box-grid-item rs-flex-box-grid-item-24 rs-col rs-col-md-2"
        role="gridcell"
      >
        <div
          class="sc-bczRLJ hnZsda rs-flex-box-grid rs-flex-box-grid-top rs-flex-box-grid-start"
        >
          <div
            class="sc-gsnTZi iiSpRq rs-flex-box-grid-item rs-flex-box-grid-item-0"
          >
            <b
              class="sc-dkzDqf cybjuV"
            >
              72
            </b>
            <b>
              °F
            </b>
          </div>
          <div
            class="sc-gsnTZi iiSpRq rs-flex-box-grid-item rs-flex-box-grid-item-0"
          >
            <b
              class="sc-dkzDqf cybjuV"
            >
              91
            </b>
            <b>
              °F
            </b>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rs-flex-box-grid-item rs-flex-box-grid-item-24 rs-col rs-col-md-2"
      role="gridcell"
    >
      <div
        class="sc-bczRLJ hnZsda rs-flex-box-grid rs-flex-box-grid-top rs-flex-box-grid-start"
      >
        <div
          class="sc-gsnTZi iiSpRq rs-flex-box-grid-item rs-flex-box-grid-item-0"
        >
          <b
            class="sc-dkzDqf cybjuV"
          >
            72
          </b>
          <b>
            °F
          </b>
        </div>
        <div
          class="sc-gsnTZi iiSpRq rs-flex-box-grid-item rs-flex-box-grid-item-0"
        >
          <b
            class="sc-dkzDqf cybjuV"
          >
            91
          </b>
          <b>
            °F
          </b>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
